### PR TITLE
feat: strapi blog integration

### PR DESCRIPTION
**Pull request by gluneau**
- Strapi blog integration
- Table of content generation with anchors
  - no header in the markdown, no toc.

Large multi level toc:
image

No toc:
image

Flat toc:
image

### CODE REVIEW

File: nuxt.config.ts [hunk 1]
```diff
@@ -47,8 +47,8 @@ export default defineNuxtConfig({
       community: {
         httpEndpoint: "https://community.astar.network/graphql",
       },
-      subsocial: {
-        httpEndpoint: "https://squid.subsquid.io/subsocial/graphql",
+      strapi: {
```
Comment (gluneau) [inline nuxt.config.ts:50]: to be removed before merging

File: components/blog/index.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// @ts-ignore
+import plugin from "markdown-it-named-headings";
+
+export async function getPosts(
+  filters: string = "",
+  pagination: string = "page: 1, pageSize: 100"
+) {
+  const { locale } = useI18n();
+  const md = new MarkdownIt().use(plugin);
+  const strapiUrl = "https://community-stg.astar.network";
```
Comment (gluneau) [inline components/blog/index.ts:12]: Will be updated  before merge and going to production

File: nuxt.config.ts [hunk 3]
```diff
@@ -45,15 +45,12 @@ export default defineNuxtConfig({
   apollo: {
     clients: {
       community: {
-        httpEndpoint: "https://community.astar.network/graphql",
-      },
-      subsocial: {
-        httpEndpoint: "https://squid.subsquid.io/subsocial/graphql",
+        httpEndpoint: "https://community-stg.astar.network/graphql",
```
Comment (gluneau) [inline nuxt.config.ts:48]: Will be updated  before merge and going to production

File: nuxt.config.ts [hunk 4]
```diff
@@ -45,15 +45,12 @@ export default defineNuxtConfig({
...
       },
     },
   },
   strapi: {
-    url: "https://community.astar.network",
+    url: "https://community-stg.astar.network",
```
Comment (gluneau) [inline nuxt.config.ts:53]: Will be updated  before merge and going to production

File: components/home/LatestNews.vue [hunk 5]
```diff
@@ -35,44 +35,10 @@
...
-      ...item,
-      image: item.image
-        ? "https://ipfs.subsocial.network/ipfs/" + item.image
-        : "/images/blog/placeholder.webp",
-      publishedDate: formattedDate,
-      slug: lowercaseSlug,
-    };
-  }
-);
+const pagination = "limit: 4";
```
Comment (ayumitk) [inline components/home/LatestNews.vue:40]:
```suggestion
const pagination = "limit: 12";
```
Please set this limit to 12. The blog posts are displayed as a carousel.
Screenshot 2024-03-21 at 9 28 44 AM

File: components/japan/Blog.vue [hunk 6]
```diff
@@ -16,47 +16,12 @@
 </template>
 
 <script setup lang="ts">
-import gql from "graphql-tag";
+import { getPosts } from "@/components/blog";
```
Comment (ayumitk) [inline components/japan/Blog.vue:19]:
```
const pagination = "limit: 3";
```
Please add limit 3 here.
Screenshot 2024-03-21 at 9 55 19 AM

**Review by ayumitk — APPROVED** [review]
LGTM! I'll add styling next week

**Review by Kahonnohak — APPROVED** [review]
LGTM!

**Pull request merged**